actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -278,12 +278,21 @@ def server_software
     def raw_post
       unless @env.include? 'RAW_POST_DATA'
         raw_post_body = body
-        @env['RAW_POST_DATA'] = raw_post_body.read(content_length)
+        @env['RAW_POST_DATA'] =
+          if chunked?
+            raw_post_body.read
+          else
+            raw_post_body.read(content_length)
+          end
         raw_post_body.rewind if raw_post_body.respond_to?(:rewind)
       end
       @env['RAW_POST_DATA']
     end
 
+    def chunked?
+      headers['Transfer-Encoding'].try(:downcase) == 'chunked'
```

Comment:
At the Rails organization we like relying on Ruby's semantics and don't particularly care for constructs like `!!`. So if we can make due without doing that then :+1: